### PR TITLE
Use ARGs in FROM of test Dockerfiles

### PR DESCRIPTION
**Pull request by MichaelSimons**
This wasn't possible before because the version of docker on the CI machines didn't support referencing ARGs in the FROM instruction.  This change cleans up the code a little and the tests will run a little faster.

### CODE REVIEW

File: test/Microsoft.DotNet.Docker.Tests/Dockerfile.linux.testapp.selfcontained [hunk 1]
```diff
@@ -1,4 +1,5 @@
-FROM {base_image}
+ARG base_image
+FROM $base_image
```
Comment (ravimeda) [inline test/Microsoft.DotNet.Docker.Tests/Dockerfile.linux.testapp.selfcontained]: Clarification - `$base_image` vs. `${base_image}` what is the purpose of the curly brackets? Is there a risk by not including them. For example, I would prefer to have quotes around a variable that contains a path `"$programfiles%"`.

File: test/Microsoft.DotNet.Docker.Tests/ImageTests.cs [hunk 2]
```diff
@@ -75,22 +75,21 @@ public void VerifyImages(ImageDescriptor imageDescriptor)
...
             string sdkImage = GetDotNetImage(
                 imageDescriptor.SdkVersion, DotNetImageType.SDK, imageDescriptor.OsVariant);
 
             DockerHelper.Build(
                 dockerfile: $"Dockerfile.{DockerHelper.DockerOS.ToLower()}.testapp",
-                fromImage: sdkImage,
                 tag: appSdkImage,
-                buildArgs: buildArgs);
+                fromImage: sdkImage,
+                buildArgs: buildArgs.ToArray());
```
Comment (ravimeda) [inline test/Microsoft.DotNet.Docker.Tests/ImageTests.cs]: `ToArray()` is that better (size/time) as opposed to passing the list.

**Comment by ravimeda** [comment]
Looks good. Please rebase.

File: test/Microsoft.DotNet.Docker.Tests/Dockerfile.linux.testapp.selfcontained (hunk 1 above)
Comment (MichaelSimons) [reply test/Microsoft.DotNet.Docker.Tests/Dockerfile.linux.testapp.selfcontained]: $base_image is they way you [reference an ARG within a Dockerfile](https://docs.docker.com/engine/reference/builder/#using-arg-variables).  Before this change {} was the convention we used to search for strings to replace prior to building the Dockerfile.  Your concern on spaces in not applicable as the FROM image cannot contain spaces.

Comment (ravimeda) [reply test/Microsoft.DotNet.Docker.Tests/Dockerfile.linux.testapp.selfcontained]: Got it. Thanks.

File: test/Microsoft.DotNet.Docker.Tests/ImageTests.cs (hunk 2 above)
Comment (MichaelSimons) [reply test/Microsoft.DotNet.Docker.Tests/ImageTests.cs]: buildArgs is a params therefore it has to be an array

**Pull request merged**